draft-ietf-wish-whep.md

```diff
@@ -633,6 +633,25 @@ data: /DA8AAAAAAAAAP///wb+06ACpQAmAiRDVUVJAACcHX//AACky4AMEERJU0NZTVdGMDQ1MjAwME
 ~~~~~
 {: title="scte35 example event"}
 
+### Events over WebRTC datachannel
+
+In some architectures sending events over a long polled HTTP request from a server that isn't the media server may be costly, such servers may support events over the webRTC datachannel.
+The protocol is identical to the above, with the following exceptions:
+
+- If both the SDP offer and the SDP answer contain an application section this indicates that Datachannel events are supported.
+~~~~~
+m=application 9 UDP/DTLS/SCTP webrtc-datachannel
+~~~~~
+- If the client then opens a reliable datachannel with a label that matches the URI of the sse Location: the server MUST send events as messages over this datachannel in the format described above.
+- The client may then listen for events on the datachannel instead of using the (similar) server-sent-events API
+
+This option can reduce resource consumption in 2 ways:
+- internal message passing within the server cluster since the events typically originate in the media server not the web server
+- port consumption, since the datachannel is multiplexed on the same port tuple as the media.
+
+It is also likely to be more timely than long polling.
```

Comment:
see the other comment about rewording "long polling"

Reply:
The problem I have with SSE is exactly that the http request often _does_ terminate and reconnect - because something in the network dropped the NAT/TCP/TLS/HTTP1.0/reverseproxy connection. I think the SSE spec mentions long polling too. But I'm happy to change the wording if it helps clarity or is a distraction.

I hadn't considered mix/match event types over the 2 different transports. I thought of them as mutually exclusive, but if you think it useful I am happy to add some words...

I don't think  we have to change the POST URI - it is just an example, not normative - but it might be clearer if we did.

Reply:
I hadn't considered mix/match event types over the 2 different transports. I thought of them as mutually exclusive, but if you think it useful please add some words...

Reply:
I can't see any mentioned of long polling in the spec - https://html.spec.whatwg.org/multipage/server-sent-events.html

They used the term "long-lived"